### PR TITLE
CAD-3638: extended trace-forward library.

### PR DESCRIPTION
**Pull request by denisshevchenko**
Since both `trace-forward` and `datapoint-forward` specify node-specific protocols, it's better to combine them into one package. This package will be used by `trace-dispatcher` library for forwarding `TraceObject`s and `DataPoint`s from the node to external acceptor app (for example, `cardano-tracer` or `cardano-wallet`).

### CODE REVIEW

File: trace-forward/src/Trace/Forward/Utils/DataPoint.hs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  atomically $ do
+    modifyTVar' dataPointsNames $ const dpNames -- Fill the names of 'DataPoint's we need.
+    modifyTVar' askDataPoints $ const True      -- Ask them! The flag for acceptor's part
+                                                -- of the protocol, it's initiate the request.
+  -- Since the acceptor's part of the protocol already sent the request,
+  -- we are waiting for reply: currently 'dataPointsReply' is empty,
+  -- so we are stuck on 'retry'.
+  --
+  -- Unfortunately, it's possible that 'dataPointsReply' won't be filled by 'DataPointValues',
+  -- because of some error (for example, network problems). To prevent an ifinite 'retry',
```
Comment (newhoggy) [inline trace-forward/src/Trace/Forward/Utils/DataPoint.hs:107]: Typo `ifinite`

**Review by newhoggy — APPROVED** [review]
LGTM

**Review by deepfire — APPROVED** [review]
I think we're clear to merge.

**Comment by denisshevchenko** [comment]
bors cancel

**Comment by denisshevchenko** [comment]
bors merge

**Pull request merged**